code/lib/core-server/src/utils/StoryIndexGenerator.ts

```diff
@@ -307,29 +317,27 @@ export class StoryIndexGenerator {
         };
       });
 
-    const { autodocs } = this.options.docs;
     // We need a docs entry attached to the CSF file if either:
-    //  a) autodocs is globally enabled
-    //  b) we have autodocs enabled for this file
-    //  c) it is a stories.mdx transpiled to CSF
+    //  a) we have autodocs enabled for this file
+    //  b) it is a stories.mdx transpiled to CSF
     const hasAutodocsTag = entries.some((entry) => entry.tags.includes(AUTODOCS_TAG));
     const isStoriesMdx = entries.some((entry) => entry.tags.includes(STORIES_MDX_TAG));
-    const createDocEntry =
-      autodocs === true || (autodocs === 'tag' && hasAutodocsTag) || isStoriesMdx;
+    const createDocEntry = (hasAutodocsTag && this.options.docs.autodocs !== false) || isStoriesMdx;
 
     if (createDocEntry && this.options.build?.test?.disableAutoDocs !== true) {
       const name = this.options.docs.defaultName ?? 'Docs';
       const { metaId } = indexInputs[0];
       const { title } = entries[0];
-      const metaTags = indexInputs[0].metaTags || [];
       const id = toId(metaId ?? title, name);
+      const tags = combineTags(...projectTags, ...(indexInputs[0].tags ?? []));
```

Comment:
I think this reverses the fix you did in https://github.com/storybookjs/storybook/pull/25537

With this change, the autodocs entry will again inherit all the tags from the primary story.

I'm not sure there's a clean workaround for it though.